File: app/src/main/java/com/android/ashwiask/tvmaze/di/ActivityBuildersModule.java
package com.android.ashwiask.tvmaze.di;

import com.android.ashwiask.tvmaze.favorite.FavoriteShowsActivity;
import com.android.ashwiask.tvmaze.home.HomeActivity;
import com.android.ashwiask.tvmaze.shows.AllShowsActivity;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;

@Module
abstract class ActivityBuildersModule {
    @ContributesAndroidInjector
    abstract HomeActivity bindHomeActivity();

    @ContributesAndroidInjector
    abstract AllShowsActivity bindAllShowsActivity();

    @ContributesAndroidInjector
    abstract FavoriteShowsActivity bindFavoriteShowsActivity();
}

File: app/src/main/java/com/android/ashwiask/tvmaze/di/ActivityBuildersModule.kt
package com.android.ashwiask.tvmaze.di

import com.android.ashwiask.tvmaze.favorite.FavoriteShowsActivity
import com.android.ashwiask.tvmaze.home.HomeActivity
import com.android.ashwiask.tvmaze.shows.AllShowsActivity

import dagger.Module
import dagger.android.ContributesAndroidInjector

@Module
abstract class ActivityBuildersModule {
    @ContributesAndroidInjector
    abstract fun bindHomeActivity(): HomeActivity

    @ContributesAndroidInjector
    abstract fun bindAllShowsActivity(): AllShowsActivity

    @ContributesAndroidInjector
    abstract fun bindFavoriteShowsActivity(): FavoriteShowsActivity
}
